bindings/pyroot/pythonizations/src/PyROOTModule.cxx

```diff
@@ -60,6 +61,69 @@ PyObject *RegisterExecutorAlias(PyObject * /*self*/, PyObject *args)
    Py_RETURN_NONE;
 }
 
+namespace Internal {
+/// \brief A PyObject wrapper to track reference counting of external objects
+///
+/// This wrapper can be useful in shared ownership scenarios when a C++ object
+/// is created on the Python side and there is no easy way to track its ownership
+/// on the C++ side. If multiple instances of this class are given the same
+/// Python proxy, they will increase/decrease its reference counting following
+/// Python rules and ensure proper destruction of the underlying C++ object when
+/// no other Python objects are referencing it.
+class PyObjRefCountWrapper final {
```

Comment:
```suggestion
class PyObjRefCount final {
```
As said in the other comment: the `std::any` is the wrapper, so maybe not use "Wrapper" in this class name?